biscuit/src/Auth/Biscuit/Datalog/Executor.hs

```diff
@@ -158,13 +158,17 @@ keepAuthorized (FactGroup facts) authorizedOrigins =
   let isAuthorized k _ = k `Set.isSubsetOf` authorizedOrigins
    in FactGroup $ Map.filterWithKey isAuthorized facts
 
-keepAuthorized' :: Natural -> FactGroup -> Set EvalRuleScope -> Natural -> FactGroup
-keepAuthorized' blockCount factGroup trustedBlocks currentBlockId =
+keepAuthorized' :: Bool -> Natural -> FactGroup -> Set EvalRuleScope -> Natural -> FactGroup
+keepAuthorized' allowPreviousInAuthorizer blockCount factGroup trustedBlocks currentBlockId =
   let scope = if null trustedBlocks then Set.singleton OnlyAuthority
                                     else trustedBlocks
       toBlockIds = \case
         OnlyAuthority    -> Set.singleton 0
-        Previous         -> Set.fromList [0..currentBlockId]
+        Previous         -> if allowPreviousInAuthorizer || currentBlockId < blockCount
```

Comment:
Is arithmetic on block ids safe?

Reply:
Yes: the block id is the index in the block list, so range operations exactly match the semantics of `Previous`